packages/coinstac-ui/app/render/state/ducks/auth.js

```diff
@@ -145,28 +157,30 @@ export const signUp = applyAsyncLoading(user => (dispatch, getState) => axios.po
     return initCoreAndSetToken(user, data, appDirectory, dispatch);
   })
   .catch((err) => {
-    if (err.response && err.response.data && (err.response.data.message === 'Username taken'
-          || err.response.data.message === 'Email taken')) {
-      dispatch(setUser({ ...INITIAL_STATE, error: err.response.data.message }));
+    const { message } = getErrorDetail(err);
+    if (message === 'Username taken' || message === 'Email taken') {
```

Comment:
These string comparisons are not really good also. What about using a 409 status code on these cases?

Reply:
@rochaeb , The api is returning these messages using Boom.badRequest and error code is 401.
Can we use 401 instead of 409?

Reply:
Yeah. No problem. It's just that 409 is better in this case...